docs/adr/0003-event-broker-for-subscription-platform.md

```diff
@@ -0,0 +1,108 @@
+* Difficulty of schema migrations in existing FxA microservices
+* Suitability of existing FxA databases for large-scale event storage
+
+## Considered Options
+
+* Implementing RP notification as a FxA Auth Service feature
+    * A. Storing webhook and login activity in FxA Auth Service
+* Implementing RP notification as a new FxA Event Broker service
+    * B. Storing webhook and login activity in FxA Auth Service
+    * C. Storing webhook and login activity in FxA Event Broker
+
+## Decision Outcome
+
+Chosen Option: C. Implementing RP notification as a new FxA Event Broker service
+with webhook and login activity stored in FxA Event Broker, because
+
+* Less subscription platform timeline risk to store data in new database vs.
+  modify existing FxA OAuth database.
+* Storing events at scale has database requirements that don't fit in well with
+  the limitations with MySQL encountered in FxA Auth/OAuth.
+* Having FxA Auth be the only store of which RP to notify would require each
+  notification to also include what RPs to notify, increasing the load on the
+  FxA Auth database.
+
+## Pros and Cons of the Options
+
+### A. Implementing RP notification as a FxA Auth Service feature w/webhook & login activity
+
+* Description
+  * FxA Auth Service has additional feature code added to perform webhook
+    deliveries to RPs.
+  * FxA Auth Service stores login activity and webhook mapping to OAuth Clients
+* Pros
+  * Auth service has direct access to subscription capability mapping and
+    OAuth webhook URLs.
+* Cons
+  * Event logs for each user are not well suited for existing MySQL database.
+  * Additional database schema migrations needed.
+  * Later features needing to query for user events could impact FxA Auth Service
+    availability.
+
+### B. Implementing RP notification as a new FxA Event Broker service w/webhook & login activity in FxA Auth Service
+
+* Description
+  * FxA Event Broker performs webhook deliveries to RPs.
+  * FxA Auth Service stores login activity and webhook mapping to OAuth Clients
+  * FxA Auth Server includes RPs that should be notified in the relevant
+    Service Notifications.
+  * FxA Event Broker caches new OAuth /clients request that indicates
+    subscription capabilities per RP.
+* Pros
+  * Division of event storing and distribution to a service that can be built
+    specifically for that function.
+* Cons
+  * Queries to FxA Auth Service required for each message to determine whether
+    to deliver to an RP could impact FxA Auth Service availability.
+  * FxA OAuth server should still have a record of which RPs a user has logged
+    into so a later issue to add this is still needed.
+
+### C. Implementing RP notification as a new FxA Event Broker service w/webhook & login activity in FxA Event Broker
+
+* Description
+  * FxA Event Broker performs webhook deliveries to RPs.
+  * FxA Event Broker caches new OAuth /clients request that indicates
+    subscription capabilities per RP.
```

Comment:
One thing that wasn't obvious to me from reading the doc, is where the new RP metadata lives.  IIRC you propose to store both the webhook URI and the subscription details in the auth server, and have the event broker fetch them with some caching behaviour. It may be worth trying to explain that in a little more detail.

Reply:
For the MVP, I was planning to store the webhook URI and last login in the event broker, as its a smaller unit of work to plan for our sprints. We can revisit if that's the shipping version later depending on if other tasks complete early (:crossed_fingers:), but given it involves a db migration to add a table and some other chunks, its definitely more work than it would be to add it in the event broker under our timelines.